ceilometer_publisher_zeromq/queue.py

```diff
@@ -0,0 +1,61 @@
+from ceilometer import publisher
+from ceilometer.openstack.common.gettextutils import _
+from ceilometer.openstack.common import log
+from oslo.config import cfg
+import json
+import zmq
+
+LOG = log.getLogger(__name__)
+
+OPTS = [
+    cfg.StrOpt('publisher_zeromq_host',
+               default='localhost'),
+    cfg.IntOpt('publisher_zeromq_port',
+               default=8282),
+]
+
+cfg.CONF.register_opts(OPTS)
+
+class QueuePublisher(publisher.PublisherBase):
+    """Republishes all received samples to a collector via ZeroMQ"""
+
+    def __init__(self, parsed_url):
+        LOG.info("Queue Publisher starting up")
+        super(QueuePublisher, self).__init__(parsed_url)
+        self.context = None
+        self.socket  = None
+        self.connect()
+
+    def reconnect(self):
+        """Terminates existing context then reconnects to the collector"""
+        self.socket.close()
+        self.context.term()
+        self.connect()
+
+    def connect(self):
+        """connects to the collector"""
+        self.context = zmq.Context()
+        self.socket  = self.context.socket(zmq.REQ)
+        self.socket.connect("tcp://%s:%d" % (cfg.CONF.publisher_zeromq_host, cfg.CONF.publisher_zeromq_port))
+
+    def publish_sample(self, message):
```

Comment:
@fractalcat it seems that the agent catches it (or something) logs it, then continues.

2015-02-27 00:43:57.990 8775 ERROR ceilometer.pipeline [-] Pipeline meter_sink: Continue after error from publisher <ceilometer_publisher_zeromq.queue.QueuePublisher object at 0x2236a50>